### ISSUE TITLE
perf: shared albums are slow

### ISSUE THREAD

**Issue by pulsejet** [pulsejet/memories#753]
Reported at Discord: https://discord.com/channels/1100157156742926406/1100158171621904567/1134462269523230740

```
public shared album
everything works very well and smooth for logged users
but UI and thumbnail loading gets extremely slow when accessing the album using the public link
as a logged in user, even clicking on the share button causes a loading window for a minute or so, and in the server I can see this
```

![image](https://github.com/pulsejet/memories/assets/10709794/9edb69e9-5385-4167-828b-f2210cd1d13a)

**Comment by jmlgo** [comment]
more info
Nextcloud 27.0.1.2

clicking on the share button from an album causes a modal to freeze loading for a minute or show, with postgres showing a high CPU consumption
![image](https://github.com/pulsejet/memories/assets/16480649/26af5ea6-d1bc-4d81-85b2-0ff726023672)

using a public link to the folder (containing all the pics in the album) shows the same behavior.. high cpu usage and lazzy thumbnail loading
previews seems to be well generated, since Memories app loads instantly the whole album

**Comment by johnhelt** [comment]
not seeing this on my docker-compose setup (nextcloud 25.0.6 (php-fpm), using mariadb and redis) running memories 5.2.1. I checked through incognito and public link on memories behind a vpn (no local ethernet traffic), with lightning fast speeds.

**Comment by jmlgo** [comment]
@johnhelt how large is your folder?
just realised that anything related with public shares is extremly slow in my server, causing postgresql db to eat all the CPU. I've tried disabling suspicious login app, and bruteforce protection, but still no clue about what is causing this.

**Comment by pulsejet** [comment]
This likely has to do with the fact that all photos are of the same day. Just needs more optimization or faster hardware at this point.

**Comment by jmlgo** [comment]
solved by nuking oc_filecache and rebuilding the index, something was bloating the database.
sorry for blaming Memories.

**Comment by pulsejet** [comment]
Great! I'm gonna keep this open for a different reason: the sharing dialog loads the entire album when it shouldn't. This is a bug in the Photos DAV API.

**Comment by jmlgo** [comment]
yeah I can see some delay with a 1600 pics album (nothing like before tho).
edit album button has the same delay btw.

**Comment by ChildLearningClub** [comment]
So it was the cobwebs on the database :)

**Comment by pulsejet** [comment]
This ought to be fixed on master now
